src/main/java/com/tabnine/binary/fetch/BootstrapperSupport.java

```diff
@@ -0,0 +1,73 @@
+package com.tabnine.binary.fetch;
+
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.util.text.SemVer;
+import com.tabnine.general.StaticConfig;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.prefs.Preferences;
+
+import static java.lang.String.format;
+
+public class BootstrapperSupport {
+    static Optional<BinaryVersion> bootstrap_version(LocalBinaryVersions localBinaryVersions, BinaryRemoteSource binaryRemoteSource, BinaryDownloader binaryDownloader) {
```

Comment:
Receive dependencies through the constructor.

Reply:
why? what is the advantage? this just a once off funciton and no need to construct it